google-sheet-sync/functions/index.js

```diff
@@ -69,11 +69,11 @@ exports.oauthcallback = functions.https.onRequest((req, res) => {
     // Now tokens contains an access_token and an optional refresh_token. Save them.
     if (err) {
       res.status(400).send(err);
-      return;
+      return null;
     }
-    db.ref(DB_TOKEN_PATH).set(tokens).then(
-        () => res.status(200).send('App successfully configured with new Credentials. ' +
-                                   'You can now close this page.'));
+    return db.ref(DB_TOKEN_PATH).set(tokens).then(
```

Comment:
More weird style. Using the inline syntax for lamdas over multiple lines just feels dirty. Also, don't do open parens and a new line unless you're going to do extra indents (useful only for multiple parameters)

```
return db.ref(DB_TOKEN_PATH).
  .then(() => res.status.send(
    'App path successfully configured with new Credentials. ' +
    'You can now close this page'));
```

or probably more preferred: 

```
return db.ref(DB_TOKEN_PATH).then(() => {
  return res.status.send(
    'App path successfully configured with new Credentials. ' +
    'You can now close this page')
});
```